benchmark/DriverBench/SingleDocBench.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace MongoDB\Benchmark\DriverBench;
+
+use Generator;
+use MongoDB\Benchmark\BaseBench;
+use PhpBench\Attributes as Bench;
+
+use function array_map;
+use function range;
+
+/** @see https://github.com/mongodb/specifications/blob/ddfc8b583d49aaf8c4c19fa01255afb66b36b92e/source/benchmarking/benchmarking.rst#single-doc-benchmarks */
+#[Bench\BeforeMethods('prepareDatabase')]
+final class SingleDocBench extends BaseBench
+{
+    public function prepareDatabase(): void
+    {
+        self::getCollection()->drop();
+    }
+
+    public function setupFindOneByID(): void
+    {
+        $tweet = (array) self::readJsonFile(self::TWEET_FILE_PATH);
+        $docs = array_map(fn ($id) => $tweet + ['_id' => $id], range(1, 10_000));
+        self::getCollection()->insertMany($docs);
+    }
+
+    #[Bench\BeforeMethods('setupFindOneByID')]
+    #[Bench\Revs(1)]
+    #[Bench\Iterations(10)]
+    public function benchFindOneByID(): void
+    {
+        $collection = self::getCollection();
+
+        for ($id = 1; $id <= 10_000; $id++) {
+            $collection->findOne(['_id' => $id]);
+        }
+    }
```

Comment:
Spec: https://github.com/mongodb/specifications/blob/master/source/benchmarking/benchmarking.rst#find-one-by-id

I could use the `Bench\Rev` annotation to run the test several times instead of having the loop in the test.